ingest/importer/importer.py

```diff
@@ -173,9 +174,22 @@ def __init__(self, template_mgr):
         self.template_mgr = template_mgr
         self.logger = logging.getLogger(__name__)
 
-    def do_import(self, workbook: IngestWorkbook, project_uuid=None):
+    def do_import(self, workbook: IngestWorkbook, is_update, project_uuid=None):
         registry = _ImportRegistry(self.template_mgr)
         importable_worksheets = workbook.importable_worksheets()
+        workbook_errors = []
+
+        if True in self.entities_by_uuid_existence(importable_worksheets).values():
+            if not is_update:
+                e = UnexpectedEntityUUIDFound()
+                workbook_errors.append(
```

Comment:
Could append a workbook_error for each offending worksheet. Otherwise, I think the user will only see an error for the first offending worksheet and not the others. For example, by moving all the new logic into something like...

```
def validate_uuid_columns(self, workbook, is_update):
   errors = []
   ...
   <existing logic to find unexpected/missing UUID columns>
   <build up errors list>
   ...
   return errors
```
...and then doing `workbook_errors.extend(self.validate_uuid_columns())`

Reply:
Hey @rolando-ebi,
Thanks for the review. I tried to improve it and also added 2 more unit tests for error cases.
Could you do another review, please?
Tx

Reply:
I added more improvement on the UUID validation